test/replication-luatest/gh_6033_box_promote_demote_test.lua

```diff
@@ -27,24 +27,14 @@ g.before_each(function(g)
         replication = {
```

Comment:
Lets squash commits which fix problems of the previous commits. To make them cleaner. Otherwise making a review is really hard - I review some code and then in the next commit it is just rewritten/fixed.

I will be back when the commits fixing older commits are squashed.